### PR TITLE
LXQt: update to version 2.0.0

### PR DESCRIPTION
**Pull request by romildo**
## Description of changes

Update LXQt to version [2.0.0](https://lxqt-project.org/release/2024/04/15/release-lxqt-2-0-0/)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/desktops/lxqt/compton-conf/default.nix [hunk 1]
```diff
@@ -11,7 +10,7 @@
 , gitUpdater
 }:
 
-mkDerivation rec {
+stdenv.mkDerivation rec {
```
Comment (wegank) [inline pkgs/desktops/lxqt/compton-conf/default.nix:13]: I guess this needs `wrapQtAppsHook`?

File: pkgs/desktops/lxqt/qlipper/default.nix [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+, stdenv
 , fetchFromGitHub
 , cmake
 , qtbase
 , qttools
 , gitUpdater
 }:
 
-mkDerivation rec {
+stdenv.mkDerivation rec {
```
Comment (wegank) [inline pkgs/desktops/lxqt/qlipper/default.nix:10]: Also here.

File: pkgs/desktops/lxqt/compton-conf/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/desktops/lxqt/compton-conf/default.nix:15]: Added.

File: pkgs/desktops/lxqt/qlipper/default.nix (hunk 2 above)
Comment (romildo) [reply pkgs/desktops/lxqt/qlipper/default.nix:11]: Added.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 305107` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>deepin.deepin-album</li>
    <li>deepin.deepin-camera</li>
    <li>deepin.deepin-clone</li>
    <li>deepin.deepin-image-viewer</li>
    <li>deepin.deepin-screen-recorder</li>
    <li>deepin.image-editor</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>32 packages failed to build:</summary>
  <ul>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-session</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.qt5integration</li>
  </ul>
</details>
<details>
  <summary>54 packages built:</summary>
  <ul>
    <li>CuboCore.coreterminal</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>lxqt.compton-conf</li>
    <li>lxqt.libdbusmenu-lxqt</li>
    <li>lxqt.libfm-qt</li>
    <li>lxqt.libfm-qt_1_4</li>
    <li>lxqt.liblxqt</li>
    <li>lxqt.libqtxdg</li>
    <li>lxqt.libqtxdg_3_12</li>
    <li>lxqt.libsysstat</li>
    <li>lxqt.lximage-qt</li>
    <li>lxqt.lxqt-about</li>
    <li>lxqt.lxqt-admin</li>
    <li>lxqt.lxqt-archiver</li>
    <li>lxqt.lxqt-build-tools</li>
    <li>lxqt.lxqt-build-tools_0_13</li>
    <li>lxqt.lxqt-config</li>
    <li>lxqt.lxqt-globalkeys</li>
    <li>lxqt.lxqt-menu-data</li>
    <li>lxqt.lxqt-notificationd</li>
    <li>lxqt.lxqt-openssh-askpass</li>
    <li>lxqt.lxqt-panel</li>
    <li>lxqt.lxqt-policykit</li>
    <li>lxqt.lxqt-powermanagement</li>
    <li>lxqt.lxqt-qtplugin</li>
    <li>lxqt.lxqt-qtplugin_1_4</li>
    <li>lxqt.lxqt-runner</li>
    <li>lxqt.lxqt-session</li>
    <li>lxqt.lxqt-sudo</li>
    <li>lxqt.lxqt-themes</li>
    <li>lxqt.obconf-qt</li>
    <li>lxqt.pavucontrol-qt</li>
    <li>pcmanfm-qt (lxqt.pcmanfm-qt)</li>
    <li>lxqt.qlipper</li>
    <li>lxqt.qps</li>
    <li>lxqt.qterminal</li>
    <li>lxqt.qtermwidget</li>
    <li>lxqt.qtxdg-tools</li>
    <li>lxqt.screengrab</li>
    <li>lxqt.xdg-desktop-portal-lxqt</li>
    <li>virt-manager-qt</li>
  </ul>
</details>

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 305107` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>deepin.deepin-clone</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>86 packages built:</summary>
  <ul>
    <li>CuboCore.coreterminal</li>
    <li>deepin.dde-app-services</li>
    <li>deepin.dde-appearance</li>
    <li>deepin.dde-calendar</li>
    <li>deepin.dde-clipboard</li>
    <li>deepin.dde-control-center</li>
    <li>deepin.dde-control-center.dev</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-device-formatter</li>
    <li>deepin.dde-dock</li>
    <li>deepin.dde-dock.dev</li>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-polkit-agent</li>
    <li>deepin.dde-session</li>
    <li>deepin.dde-session-shell</li>
    <li>deepin.dde-session-shell.dev</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-widgets</li>
    <li>deepin.deepin-calculator</li>
    <li>deepin.deepin-compressor</li>
    <li>deepin.deepin-draw</li>
    <li>deepin.deepin-editor</li>
    <li>deepin.deepin-movie-reborn</li>
    <li>deepin.deepin-movie-reborn.dev</li>
    <li>deepin.deepin-music</li>
    <li>deepin.deepin-picker</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-screensaver</li>
    <li>deepin.deepin-shortcut-viewer</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deepin.deepin-terminal</li>
    <li>deepin.deepin-turbo</li>
    <li>deepin.deepin-voice-note</li>
    <li>deepin.dpa-ext-gnomekeyring</li>
    <li>deepin.dtkdeclarative</li>
    <li>deepin.dtkdeclarative.dev</li>
    <li>deepin.dtkdeclarative.doc</li>
    <li>deepin.dtkgui</li>
    <li>deepin.dtkgui.dev</li>
    <li>deepin.dtkgui.doc</li>
    <li>deepin.dtkwidget</li>
    <li>deepin.dtkwidget.dev</li>
    <li>deepin.dtkwidget.doc</li>
    <li>deepin.qt5integration</li>
    <li>lxqt.compton-conf</li>
    <li>lxqt.libdbusmenu-lxqt</li>
    <li>lxqt.libfm-qt</li>
    <li>lxqt.libfm-qt_1_4</li>
    <li>lxqt.liblxqt</li>
    <li>lxqt.libqtxdg</li>
    <li>lxqt.libqtxdg_3_12</li>
    <li>lxqt.libsysstat</li>
    <li>lxqt.lximage-qt</li>
    <li>lxqt.lxqt-about</li>
    <li>lxqt.lxqt-admin</li>
    <li>lxqt.lxqt-archiver</li>
    <li>lxqt.lxqt-build-tools</li>
    <li>lxqt.lxqt-build-tools_0_13</li>
    <li>lxqt.lxqt-config</li>
    <li>lxqt.lxqt-globalkeys</li>
    <li>lxqt.lxqt-menu-data</li>
    <li>lxqt.lxqt-notificationd</li>
    <li>lxqt.lxqt-openssh-askpass</li>
    <li>lxqt.lxqt-panel</li>
    <li>lxqt.lxqt-policykit</li>
    <li>lxqt.lxqt-powermanagement</li>
    <li>lxqt.lxqt-qtplugin</li>
    <li>lxqt.lxqt-qtplugin_1_4</li>
    <li>lxqt.lxqt-runner</li>
    <li>lxqt.lxqt-session</li>
    <li>lxqt.lxqt-sudo</li>
    <li>lxqt.lxqt-themes</li>
    <li>lxqt.obconf-qt</li>
    <li>lxqt.pavucontrol-qt</li>
    <li>lxqt.pcmanfm-qt</li>
    <li>lxqt.qlipper</li>
    <li>lxqt.qps</li>
    <li>lxqt.qterminal</li>
    <li>lxqt.qtermwidget</li>
    <li>lxqt.qtxdg-tools</li>
    <li>lxqt.screengrab</li>
    <li>lxqt.xdg-desktop-portal-lxqt</li>
    <li>virt-manager-qt</li>
  </ul>
</details>

**Comment by wegank** [comment]
I can confirm that LXQt 2.0.0 works fine, while `lxqt-archiver` lacks the Wayland backend...

**Comment by wegank** [comment]
> Not sure what to do here: if `qtwayland` isn't added to the dependencies for each Qt 6 app, the apps will regress to using Xwayland on Wayland. However, it's possible this will eventually get fixed by #269674 instead, rendering this superfluous. Maybe worth it just to make sure there's no regression for now?

I'm sure this PR will land on master before #269674, which targets staging and has been sitting for months. So I suggest explicitly including `qtwayland` where necessary in this PR.

**Comment by romildo** [comment]
It seems that `lxqt-archiver-1.0.0` is run as a wayland window in labwc. It can be seen in this screenshot of the labwc window switcher.

image

**Comment by romildo** [comment]
It seems that `lxqt-archiver-1.0.0` is run as a wayland window in labwc. It can be seen in this screenshot of the labwc window switcher.

image

**Comment by wegank** [comment]
On my side, which is GNOME Wayland,

- On master:
  ```bash
  $ lxqt-archiver # runs via xwayland
  $ QT_QPA_PLATFORM=wayland lxqt-archiver # runs via wayland
  ```
- On master, with this PR applied:
  ```bash
  $ lxqt-archiver # runs via xwayland
  # qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
  $ QT_QPA_PLATFORM=wayland lxqt-archiver # doesn't run
  ```
- On master, with this PR applied and `qtwayland` added:
  ```bash
  $ lxqt-archiver # runs via wayland
  $ QT_QPA_PLATFORM=wayland lxqt-archiver # runs via wayland
  ```

**Review by wegank — APPROVED** [review]

**Pull request merged**